### PR TITLE
build/configs: Fix reset issue at imxrt flash init

### PR DESCRIPTION
**Pull request by sangam-swami**
Modify linker scripts to fix flexspi flash configuration issue

Signed-off-by: Sangamanatha <sangam.swami@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [cf9db99443a2ba8a308090ecd272fb9d492981c6] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [cf9db99443a2ba8a308090ecd272fb9d492981c6] - Code Rule Check (C++) OK.

File: build/configs/imxrt1020-evk/scripts/flash-ocram.ld [hunk 1]
```diff
@@ -134,6 +134,8 @@ SECTIONS
 		*(.data .data.*)
 		*(.gnu.linkonce.d.*)
 		CONSTRUCTORS
+		*imxrt_flexspi.o(.text*)
+        *imxrt_norflash.o(.text*)
```
Comment (sijupunnnoose) [inline build/configs/imxrt1020-evk/scripts/flash-ocram.ld]: Can you align it properly?

File: build/configs/imxrt1020-evk/scripts/kernel-space-dtcm.ld [hunk 2]
```diff
@@ -93,7 +93,7 @@ SECTIONS
   {
     _stext = ABSOLUTE(.);
     *(.vectors)
-    *(.text .text.*)
+	*(EXCLUDE_FILE(*imxrt_flexspi.o *imxrt_norflash.o) .text*)
```
Comment (sijupunnnoose) [inline build/configs/imxrt1020-evk/scripts/kernel-space-dtcm.ld]: Ditto

File: build/configs/imxrt1050-evk/scripts/kernel-space.ld [hunk 3]
```diff
@@ -93,7 +93,7 @@ SECTIONS
   {
     _stext = ABSOLUTE(.);
     *(.vectors)
-    *(.text .text.*)
+	*(EXCLUDE_FILE(*imxrt_flexspi.o *imxrt_hyperflash.o) .text*)
```
Comment (sijupunnnoose) [inline build/configs/imxrt1050-evk/scripts/kernel-space.ld]: Ditto

File: build/configs/imxrt1050-evk/scripts/kernel-space.ld [hunk 4]
```diff
@@ -134,6 +134,8 @@ SECTIONS
     *(.data .data.*)
     *(.gnu.linkonce.d.*)
     CONSTRUCTORS
+	*imxrt_flexspi.o(.text*)
```
Comment (sijupunnnoose) [inline build/configs/imxrt1050-evk/scripts/kernel-space.ld]: Ditto

File: build/configs/imxrt1050-evk/scripts/kernel-space.ld [hunk 5]
```diff
@@ -134,6 +134,8 @@ SECTIONS
...
+		*imxrt_hyperflash.o(.text*)
```
Comment (sijupunnnoose) [inline build/configs/imxrt1050-evk/scripts/kernel-space.ld]: Ditto

File: build/configs/imxrt1050-evk/scripts/kernel-space-dtcm.ld [hunk 6]
```diff
@@ -134,6 +134,8 @@ SECTIONS
     *(.data .data.*)
     *(.gnu.linkonce.d.*)
     CONSTRUCTORS
+	*imxrt_flexspi.o(.text*)
+		*imxrt_hyperflash.o(.text*)
```
Comment (sijupunnnoose) [inline build/configs/imxrt1050-evk/scripts/kernel-space-dtcm.ld]: Ditto

**Comment by seinfra** [comment]
Target : [716a21f51425870116ada8bde1a2181d2a2e0a1d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [716a21f51425870116ada8bde1a2181d2a2e0a1d] - Code Rule Check OK.

File: build/configs/imxrt1020-evk/scripts/flash-ocram.ld (hunk 1 above)
Comment (sangam-swami) [reply build/configs/imxrt1020-evk/scripts/flash-ocram.ld]: ok

File: build/configs/imxrt1050-evk/scripts/kernel-space-dtcm.ld (hunk 6 above)
Comment (sangam-swami) [reply build/configs/imxrt1050-evk/scripts/kernel-space-dtcm.ld]: ok

File: build/configs/imxrt1050-evk/scripts/kernel-space.ld (hunk 4 above)
Comment (sangam-swami) [reply build/configs/imxrt1050-evk/scripts/kernel-space.ld]: ok

File: build/configs/imxrt1050-evk/scripts/kernel-space.ld (hunk 3 above)
Comment (sangam-swami) [reply build/configs/imxrt1050-evk/scripts/kernel-space.ld]: ok

**Comment by sangam-swami** [comment]
> Fix alignment.
> How moving imxrt_flexspi & hyperflash object files fixes this? Could you brief the issue as the commit description dont say much.

Hi,
In case of XIP, flexispi and hyperflash objects run on top of flash, if we copy these in flash itself then unexpected behavior results. So these are copied to data section to execute the flash init and configure from outside

**Comment by sunghan-chang** [comment]
@sangam-swami Could you leave what you said in commit description?
> In case of XIP, flexispi and hyperflash objects run on top of flash, if we copy these in flash itself then unexpected behavior results. So these are copied to data section to execute the flash init and configure from outside

**Comment by seinfra** [comment]
Target : [7035ec6793c5900a96520ace7430e6da331d581f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [7035ec6793c5900a96520ace7430e6da331d581f] - Code Rule Check (C++) OK.

**Pull request merged**